Implemented taxi lookahead

Althar93 opened the pull request:
A first attempt at implementing #5 

The taxi lookahead is kept separate to the ground roll as suggested in the original request. 

The lookahead is derived from a blend of the rudder input & ground rotation rate in an attempt to orient the view towards the turn. To minimise the distractions that the lookahead would cause during the takeoff phase, the effect is set to fade out with increased ground speed (currently set to 20 knots which should work for most aircrafts).

I have ran a series of tests with both a set of Thrusmaster Rudder pedals as well as a regular twist-style joystick rudder control (X55 Hotas) for sensitivity/response. I experimented with the lookahead effect in isolation as well as with all other ground effects enabled on both GA planes & larger Heavy aircrafts.

The initial responses & max taxi speeds default values were set to what I considered to feel & interact nicely across a variety of aircraft types. 

Suggestions for different default values are of course welcome ; general feedback would also be much appreciated.

Thanks

Althar93 committed:
Implemented taxi lookahead

nico87 commented:
Great job! The final effect is better than what I was expecting.
I have very few considerations before merging the PR.

1. I would lower the default rudder response at about 5 or 10.
2. What do you think of adding a little bit of (configurable) forward head movement while turning? I usually tend to lean forward while turning before the Cessnas have the door strut right between your eyes!
3. I had some complaints in the past about the settings window size. Maybe it's better to have two columns where we can put the "Look ahead effect", "Ground roll vibrations" and "Taxi lookahead" packed together. So, Look ahead effect and ground roll vibration on the same column on the left and "Taxi lookahead" on the right, taking the two rows of the other two effect. In this way we should be able to keep the current window size!

Althar93 commented:
All of the suggestions/comments above sound reasonable. I will look into making those ammendments hopefully before the end of the weekend. 

Just one question: when you say 'lean forward' do you mean leaning into the turn/lookahead direction (as if you were trying to stick your head outside) or did I misunderstand?

nico87 commented:
> Just one question: when you say 'lean forward' do you mean leaning into the turn/lookahead direction (as if you were trying to stick your head outside) or did I misunderstand?

I mean leaning toward the nose of the engine. Does not matter the direction. In this way, if you lean forward and turn your head, you can have a better view of the outside.

Althar93 committed:
Added forward/upward lean motion to the taxi lookahead.
Updated/reduce the default rudder response for the taxi lookahead.
Reworked/compressed the menu with the taxi lookahead now in a second column alongside groundroll & lookahead.

Althar93 commented:
Thanks for clarifying. 

I have just submitted some changes as recommended above, namely : 
* Default internal rudder response updated to match what was previously 10% (the slider still displays 25% for convenience)
* Added basic rudder-based lean behaviour. The camera will move towards the nose of the aircraft with a slight upward motion. The default response was set based on a range of GA & Heavy cockpits, although as the boundaries are unknown, this setting may vary from cockpit to cockpit.
* Compressed the menu into two columns (retaining the original menu size) with ground roll & lookahead on the left hand side & taxi look ahead on its own on the right hand side.

nico87 commented:
Great job!
I've merged this into master and I'm going to make a build of HS 1.6.

Please contact me at info[at]simcoders[dot]com.

Thank you!
